operator/bucketcontroller/delete.go

```diff
@@ -28,6 +31,7 @@ func (p *ProvisioningPipeline) Delete(ctx context.Context, mg resource.Managed)
 				"delete all objects", p.deleteAllObjects,
 			),
 			pipe.NewStep("delete bucket", p.deleteS3Bucket),
+			pipe.NewStep("clean credential secret", p.cleanCredentialSecret),
```

Comment:
As mentioned before, we shouldn't mess with a secret that is managed by another controller